drivers/kw41zrf/kw41zrf_netdev.c

```diff
@@ -0,0 +1,1256 @@
+                               *((const netopt_enable_t *)value));
+            res = len;
+            break;
+
+        case NETOPT_RX_END_IRQ:
+            if (len != sizeof(const netopt_enable_t)) {
+                res = -EOVERFLOW;
+                break;
+            }
+            kw41zrf_set_option(dev, KW41ZRF_OPT_TELL_RX_END,
+                               *((const netopt_enable_t *)value));
+            res = len;
+            break;
+
+        case NETOPT_TX_START_IRQ:
+            if (len != sizeof(const netopt_enable_t)) {
+                res = -EOVERFLOW;
+                break;
+            }
+            kw41zrf_set_option(dev, KW41ZRF_OPT_TELL_TX_START,
+                               *((const netopt_enable_t *)value));
+            res = len;
+            break;
+
+        case NETOPT_TX_END_IRQ:
+            if (len != sizeof(const netopt_enable_t)) {
+                res = -EOVERFLOW;
+                break;
+            }
+            kw41zrf_set_option(dev, KW41ZRF_OPT_TELL_TX_END,
+                               *((const netopt_enable_t *)value));
+            res = len;
+            break;
+
+        case NETOPT_CSMA_RETRIES:
+            if (len != sizeof(uint8_t)) {
+                res = -EOVERFLOW;
+                break;
+            }
+            dev->csma_max_backoffs = *((const uint8_t*)value);
+            res = len;
+            break;
+
+        case NETOPT_CSMA_MAXBE:
+            if (len != sizeof(uint8_t)) {
+                res = -EOVERFLOW;
+                break;
+            }
+            dev->csma_max_be = *((const uint8_t*)value);
+            res = len;
+            break;
+
+        case NETOPT_CSMA_MINBE:
+            if (len != sizeof(uint8_t)) {
+                res = -EOVERFLOW;
+                break;
+            }
+            dev->csma_min_be = *((const uint8_t*)value);
+            res = len;
+            break;
+
+        case NETOPT_RETRANS:
+            if (len != sizeof(uint8_t)) {
+                return -EOVERFLOW;
+            }
+            dev->max_retrans = *((const uint8_t *)value);
+            res = len;
+            break;
+
+        default:
+            break;
+    }
+
+    bool put_to_sleep_when_done = false;
+
+    if (kw41zrf_is_dsm()) {
+        /* Transceiver is in deep sleep mode, check if setting the option
+         * requires the radio powered on */
+        switch (opt) {
+            case NETOPT_AUTOACK:
+            case NETOPT_ACK_REQ:
+            case NETOPT_PROMISCUOUSMODE:
+            case NETOPT_RX_START_IRQ:
+            case NETOPT_CSMA:
+            case NETOPT_ADDRESS:
+            case NETOPT_ADDRESS_LONG:
+            case NETOPT_NID:
+            case NETOPT_CHANNEL:
+            case NETOPT_TX_POWER:
+            case NETOPT_CCA_THRESHOLD:
+            case NETOPT_CCA_MODE:
+                kw41zrf_set_power_mode(dev, KW41ZRF_POWER_IDLE);
+                DEBUG("[kw41zrf] Wake to set %s\n", netopt2str(opt));
```

Comment:
Shouldn't this be enclosed in an `#ifdef` to ensure that this debug call is only performed when the `netopt` module is enabled? Otherwise the build fails if `ENABLE_DEBUG` is set and the application using this doesn't depend on the netopt module.

Reply:
Thanks, will address this

Reply:
addressed